### PR TITLE
Update to recent xarray and pandas

### PR DESCRIPTION
**Pull request by bdestombe**

### CODE REVIEW

**Comment by bdestombe** [comment]
See #167

**Comment by bdestombe** [comment]
@BSchilperoort it doesn't seem to work too well with python 3.8..

**Comment by BSchilperoort** [comment]
> @BSchilperoort it doesn't seem to work too well with python 3.8. I am okay with dropping support for 3.8 and including support for 3.11. What do you think?

That's fine with me. It looks like 3.8 fails as xarray isn't developing for 3.8 anymore: https://github.com/pydata/xarray/blob/971be103d6376d6572d1f12d32526f12f07ae2c7/setup.cfg#L76C13-L76C13

**Comment by bdestombe** [comment]
There are currently no wheels built for netCDF4 on Python 3.11

**Review by BSchilperoort — APPROVED** [review]
3.11 should work if you unpin netcdf4. It is pinned, but without a comment on why...

```py
ds_resampled = DataStore(xr.Dataset(ds).resample(time="47S").mean())
```
This feels a bit clunky. I guess there was no easy way to avoid this?
Can you add the modified behavior to the changelog, and suggest users on how to modify their existing code?

Also, I am not very keen on making these changes in a patch version (i.e. 2.0.1). Even introducing breaking changes to minor versions it technically not how it should work.

File: docs/notebooks/02Common_DataStore_functions_slice_mean_max_std_resample.ipynb [hunk 1]
```diff
@@ -273,23 +273,29 @@
...
+    "# Therefore we convert it back to the dtscalibration Datastore type.\n",
+    "from dtscalibration import DataStore\n",
+    "ds_resampled2 = DataStore(ds_xarray)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the resample function from dtscalibration has been removed in v2.0.1. The above example works versions from before 2.0.1 as well. Starting with version 2.0.1 the `xr.Dataset(ds).resample()` may become `ds.resample()`."
```
Comment (BSchilperoort) [inline docs/notebooks/02Common_DataStore_functions_slice_mean_max_std_resample.ipynb:298]:
```suggestion
    "Note that the resample function from dtscalibration has been removed in v2.1.0. The above example works versions from before 2.0.1 as well. Starting with version 2.1.0 the `xr.Dataset(ds).resample()` may become `ds.resample()`."
```

**Comment by bdestombe** [comment]
> 3.11 should work if you unpin netcdf4. It is pinned, but without a comment on why...
> 
> ```python
> ds_resampled = DataStore(xr.Dataset(ds).resample(time="47S").mean())
> ```
> 
> This feels a bit clunky. I guess there was no easy way to avoid this? Can you add the modified behavior to the changelog, and suggest users on how to modify their existing code?
> 
> Also, I am not very keen on making these changes in a patch version (i.e. 2.0.1). Even introducing breaking changes to minor versions it technically not how it should work.

So we bump up a major version?

**Comment by BSchilperoort** [comment]
> > This feels a bit clunky. I guess there was no easy way to avoid this? Can you add the modified behavior to the changelog, and suggest users on how to modify their existing code?
> > Also, I am not very keen on making these changes in a patch version (i.e. 2.0.1). Even introducing breaking changes to minor versions it technically not how it should work.
> 
> So we bump up a major version?

Technically that's what you should do. Although then we are bumping to new major versions a bit often perhaps.
We should identify everything we would still like to do before bumping to a new major version.

**Pull request merged**